test/functional/feature_txwitness.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+# Copyright (c) 2015-2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test decoding scripts via decodescript RPC command."""
+
+from test_framework.messages import CTransaction, CTxIn, CTxOut, sha256, ser_uint256, uint256_from_str
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, bytes_to_hex_str, hex_str_to_bytes
+
+from io import BytesIO
+import pprint
+
+class TxWitnessTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def test_transaction_serialization(self):
+        node = self.nodes[0]
+
+        # spend money to segwit address
+        unspent = node.listunspent()[0]
+        addr = node.getnewaddress("", "p2sh-segwit")
+        node.sendtoaddress(address=addr, amount=50, subtractfeefromamount=True)
+        node.generate(1)
```

Comment:
let's use `generatetoaddress` instead, `generate` is removed upstream. You can send funds directly to any type of address you want